core/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -66,6 +67,12 @@ public static EsThreadPoolExecutor newScaling(String name, int min, int max, lon
         return executor;
     }
 
+    public static EsThreadPoolExecutor newBoundedCached(String name, int poolSize, long keepAliveTime, TimeUnit unit, ThreadFactory threadFactory, ThreadContext contextHolder) {
+        EsThreadPoolExecutor executor = new EsThreadPoolExecutor(name, poolSize, poolSize, keepAliveTime, unit, new LinkedBlockingQueue<>(), threadFactory, new EsAbortPolicy(), contextHolder);
```

Comment:
als we can't use the EsAbortPolicy - we may never reject. We should use ForceQueuePolicy